insights/parsers/hammer_ping.py

```diff
@@ -30,69 +22,75 @@
+    'Message: 404 Resource Not Found'
+    >>> hammer.is_ok
     False
-    >>> status.response_of('foreman_tasks')
-    '1ms'
-    >>> status.response_of('candlepin')
-    '404 Resource Not Found'
+    >>> hammer.services_of_status('OK')
+    ['elasticsearch', 'foreman_tasks']
 """
 from .. import parser, Parser
 
 
 @parser("hammer_ping")
 class HammerPing(Parser):
     """
-    Read the ``hammer ping`` status and convert it to a dictionary of
-    response statuses and information.
-
-    The status and response information are available in the ``data``
-    property keyed to the name of each service, and the ``is_ok`` and
-    ``response_of`` methods are provided as conveniences.
+    Read the ``hammer ping`` status and convert it to dictionaries of
+    status and response information.
     """
-    def __len__(self):
-        """
-        (int): the number of items in the underlying data.
+    def services_of_status(self, status='ok'):
         """
-        return len(self.data)
+        List of the services in the given status.
 
-    def is_ok(self, service):
-        """
-        (bool): is the service defined in the data and is it listed as 'ok'?
-        """
-        return service in self.data and 'ok' == self.data[service]['status']
+        Arguments:
+            status (str): the status code to search for, defaulting to 'ok'.
+                The value is converted to lower case.
 
-    def response_of(self, service):
-        """
-        (str): the status of the given service, or '' if the service is not
-        in the hammer ping output.
+        Returns: List of service names having that status.
         """
-        if service not in self.data:
-            return ''
-        return self.data[service]['response']
+        status = status.lower()
+        return [svc for svc in self.service_list
+                if self.status_of_service[svc] == status]
 
     def parse_content(self, content):
-        i = 0
-        dic = {}
-        while i < len(content):
-            service_line, status_line, response_line = content[i:i + 3]
-            status = status_line.split(':', 1)[1].strip()
-            response = response_line.split(':', 2)[2].strip()
-            dic[service_line.strip(': ')] = {
-                'status': status,
-                'response': response
-            }
-            # status messages are three lines long - skip three lines.
-            i += 3
-        self.data = dic
+        self.status_of_service = {}
+        """dict: The status of each service, converted to lower case."""
+        self.response_of_service = {}
+        """dict: The response of each service, as is."""
+        self.service_list = []
+        """list: The list of service names in order."""
+        self.errors = []
+        """list: Any error messages encountered during parsing."""
+        self.is_ok = False
+        """bool: Were all services read correctly and are they all in 'ok' state?"""
```

Comment:
@xiangce could you move these docstrings to the class level so they will be included in the documentation?